chainstate/src/detail/transaction_verifier/mod.rs

```diff
@@ -256,66 +282,71 @@ impl<'a, S: BlockchainStorageRead> TransactionVerifier<'a, S> {
                             )
                         })?;
 
-                    Self::get_output_value(tx.outputs(), output_index, tx.get_id().into())
-                        .cloned()?
+                    let output_value =
+                        Self::get_output_value(tx.outputs(), output_index, tx.get_id().into())?;
+
+                    let (key, amount) = filter_for_total_inputs(output_value, &tx)?;
+                    insert_or_increase(&mut result, key, amount)?
                 }
                 common::chain::SpendablePosition::BlockReward(block_id) => {
                     let outputs = self.block_reward_outputs(block_id)?;
-                    Self::get_output_value(&outputs, output_index, (*block_id).into()).cloned()?
-                }
-            };
-            match output_amount {
-                OutputValue::Coin(output_amount) => {
-                    total = (total + output_amount)
-                        .ok_or(ConnectTransactionError::InputAdditionError)?
+                    match Self::get_output_value(&outputs, output_index, (*block_id).into())? {
+                        OutputValue::Coin(amount) => {
+                            insert_or_increase(&mut result, CoinOrTokenId::Coin, *amount)?
+                        }
+                        OutputValue::Token(_) => {
+                            return Err(ConnectTransactionError::TokensError(
+                                TokensError::BlockRewardInTokens,
+                            ));
+                        }
+                    }
                 }
             }
         }
-        Ok(total)
+        Ok(result)
+    }
+
+    fn get_paid_fee_unchecked(
```

Comment:
What is it exactly that's "unchecked" here?

Reply:
Also, since you're excluding tokens here, please throw in some "coin" in the name of the function.

Reply:
As we discussed before, functions mustn't include unclear words, or words that can confuse. Is it ok to rename it to  `get_paid_fee_coins`

Reply:
`get_total_fee()`